### DISCUSSION TITLE
Document downgrade procedure (if supported)

### DISCUSSION THREAD

**Discussion by zamazan4ik** [databendlabs/databend#9389] [Q&A] [answered]
**Summary**
After reading the section about [Upgrade](https://databend.rs/doc/deploy/upgrade/upgrade) procedure it's not clear - is the downgrade process supported or not? E.g. during the update, I found an issue with the newer version, and I want to fall back to the previous version. Could I do it? Could I do it without downtime?

**Comment by Xuanwo** [comment, marked as answer]
> is the downgrade process supported or not?

No, we don't support downgrade at this stage.

However, if no data has been written, we can switch to old versions by replacing the binary directly.

**Reply by zamazan4ik** [reply]
Could you please mention it explicitly in the documentation? Since this information will be written explicitly, system administrators will plan other ways for possible mitigations during the unsuccessful upgrade process (test before on a separate cluster, blue-green deploys with multiple clusters, etc.).

**Comment by bohutang** [comment]
> E.g. during the update, I found an issue with the newer version, and I want to fall back to the previous version. Could I do it? Could I do it without downtime?

During the update falling back to the previous version is OK.

This does not usually happen, and we generally do not change the underlying data format to cause incompatibilities, but problems may arise with two versions that span a wide range.

**Reply by zamazan4ik** [reply]
So at least during the update, if something goes wrong, the downgrade is supported? Do you write explicitly in the release notes, if the underlying data format changes incompatibly in a new version?

**Reply by bohutang** [reply]
Thanks.

We will improve the documentation, and if a breaking change occurs in a future version, we will make special instructions and tell users how to do the upgrade.
